### DISCUSSION TITLE
Show and hide weekend days for stock data

### DISCUSSION THREAD

**Discussion by dssvfx** [highfestiva/finplot#477] [Q&A]
Hi, I've been trying to understand which part of the code is responsible for deciding what timestamps to plot along the X axis. I realised that the finplot is smart enough to manage this automatically and it omits the ticks that hold no data or are not present in the dataframe. 

If I was to plot the stock data - currently the weekends are omitted automatically as well as any of the other filtered out rows from the dataframe. How can I re-enable this and show the blank spaces on the chart?

And could you please point out in the source code where exactly you have managed to resolve it automatically? I have been looking for a couple of days now going back and forth between finplot and PyQtGraph API and no luck :(
Similar to this thread:
https://groups.google.com/g/pyqtgraph/c/z2FXYJw5hck/m/1UXloGwMBgAJ

Thank you in advance!

**Comment by highfestiva** [comment]
Unfortunately I have no interest in enabling blank spaces, perhaps you can show them by inserting `NaN`s into your weekends. The reason it doesn't show up is that only days with candle data are rendered.

**Reply by dssvfx** [reply]
Hi @highfestiva , thank you for your reply. That is understandable, most of the time you don't need them to be displayed just like you wouldn't want to plot the period outside of regular trading hours too. 

But I was wondering from the technical perspective, when filtering a pandas data frame by those hours and since there is no data for the weekends anyway when plotting stock data - by default, the PlotItem fills in the gaps between those times plotting a continuous timeline along the X axis and that is what creating those gaps. 

May I ask how you got around to squishing those gaps on the X axis entirely? Is it driven by the custom implementation of the ticks of the custom Axis class that is plugged to a PlotItem or was it something else? Which part is driving the display of ticks along the X axis that allows you to drop unwanted periods entirely and avoid displaying them?

Many thanks

**Reply by highfestiva** [reply]
I don't draw anything time-based on the X-axis, I only draw one candle each for each candle I find in the data. After the fact I do a reverse lookup to see what date to show on the axis/crosshair.
